pkg/controller/ehpa/prom_adapter_configmap_controller.go

```diff
@@ -0,0 +1,120 @@
+package ehpa
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/prometheus-adapter/pkg/config"
+)
+
+// controller for configMap of prometheus-adapter
+type PromAdapterConfigMapController struct {
```

Comment:
Normally a controller should be independency. So i suggest you can implement it referring to OOMRecorder's implementation.